### DISCUSSION TITLE
zh/datasets/classify/

### DISCUSSION THREAD

**Discussion by giscus** [ultralytics/ultralytics#14975] [Docs]
# zh/datasets/classify/

了解如何构建数据集 YOLO 分类任务。详细的文件夹结构和使用示例，用于有效培训。

https://docs.ultralytics.com/zh/datasets/classify/

**Comment by MrLi-creator** [comment]
Hi, how can I change the class weight in the training of yolov8n-cls.pt?

**Reply by pderrenger** [reply]
@MrLi-creator to change the class weights in the training of `yolov8n-cls.pt`, you can modify the `class_weights` parameter in the training configuration. This parameter allows you to assign different weights to each class, which can help in handling class imbalance. Ensure you are using the latest version of the Ultralytics package for the best compatibility and features. For more detailed instructions, you can refer to the [Ultralytics documentation on classification datasets](https://docs.ultralytics.com/datasets/classify/).

**Reply by MrLi-creator** [reply]
Hi pderrenger,
Firstly, thank you for your response. I am trying to add __classw_weights__ to the __data.yaml__ file. However, it was found that the classification task cannot use the __data.yaml__ file, as you suggested in the link(https://github.com/ultralytics/ultralytics/issues/13789). So are there any other ways to modify __classw_weights in classification__, preferably with an example

**Reply by pderrenger** [reply]
@MrLi-creator hi pderrenger,

Thank you for your question. For modifying class weights in classification tasks, you can directly set the `class_weights` parameter in the training script. Unfortunately, the `data.yaml` file isn't used for classification tasks. Instead, you can specify class weights programmatically when initializing the training process. If you need further assistance, please refer to the [Ultralytics documentation on classification datasets](https://docs.ultralytics.com/datasets/classify/).

**Reply by MrLi-creator** [reply]
@pderrenger
Hi pderrenger. I have set ```class_weights``` parameter in the training script. But it report error.

```
results = model.train(data='./datasets', epochs=30, imgsz=50, device=None, batch=16,classes =[1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 3.0] )

SyntaxError: 'class_weights' is not a valid YOLO argument. Similar arguments are i.e. ['classes'].

```

**Reply by pderrenger** [reply]
Hi @MrLi-creator,

The `class_weights` parameter is not directly supported in the `train` method for YOLO classification tasks. Instead, you can handle class imbalance by adjusting the dataset or using techniques like data augmentation. If you encounter any issues, please ensure you are using the latest version of the Ultralytics package. For more detailed guidance, you can refer to the [Ultralytics documentation on classification datasets](https://docs.ultralytics.com/datasets/classify/).

**Comment by 1ZZZyyyh** [comment]
that's why? I've made my dataset as the sample.

File "/home/visitor/anaconda3/lib/python3.12/site-packages/ultralytics/engine/trainer.py", line 567, in get_dataset
    raise RuntimeError(emojis(f"Dataset '{clean_url(self.args.data)}' error ❌ {e}")) from e
RuntimeError: Dataset '/home/visitor/yolov8-tb11' error ❌ [Errno 21] Is a directory: '/home/visitor/yolov8-tb11'

**Reply by glenn-jocher** [reply]
The error indicates that the `data` argument provided points to a directory rather than a valid dataset configuration file (`.yaml`). Ensure you specify the path to a correctly formatted `data.yaml` file that defines the dataset structure. For reference, visit [Dataset Structure for YOLO Classification Tasks](https://docs.ultralytics.com/datasets/classify/). Let me know if you need further clarification!

**Reply by 1ZZZyyyh** [reply]
I want to train my model , it reports error.
yolo task=classify mode=train model=yolov8m-cls.pt data=/home/visitor/yolov8-tb11 epochs=100 imgsz=640
...
Traceback (most recent call last):
  File "/home/visitor/anaconda3/bin/yolo", line 8, in <module>
    sys.exit(entrypoint())
             ^^^^^^^^^^^^
  File "/home/visitor/anaconda3/lib/python3.12/site-packages/ultralytics/cfg/__init__.py", line 991, in entrypoint
    getattr(model, mode)(**overrides)  # default args from model
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/visitor/anaconda3/lib/python3.12/site-packages/ultralytics/engine/model.py", line 806, in train
    self.trainer.train()
  File "/home/visitor/anaconda3/lib/python3.12/site-packages/ultralytics/engine/trainer.py", line 207, in train
    self._do_train(world_size)
  File "/home/visitor/anaconda3/lib/python3.12/site-packages/ultralytics/engine/trainer.py", line 323, in _do_train
    self._setup_train(world_size)
  File "/home/visitor/anaconda3/lib/python3.12/site-packages/ultralytics/engine/trainer.py", line 264, in _setup_train
    self.amp = torch.tensor(check_amp(self.model), device=self.device)
                            ^^^^^^^^^^^^^^^^^^^^^
  File "/home/visitor/anaconda3/lib/python3.12/site-packages/ultralytics/utils/checks.py", line 707, in check_amp
    assert amp_allclose(YOLO("yolo11n.pt"), im)
                        ^^^^^^^^^^^^^^^^^^
  File "/home/visitor/anaconda3/lib/python3.12/site-packages/ultralytics/models/yolo/model.py", line 23, in __init__
    super().__init__(model=model, task=task, verbose=verbose)
  File "/home/visitor/anaconda3/lib/python3.12/site-packages/ultralytics/engine/model.py", line 146, in __init__
    self._load(model, task=task)
  File "/home/visitor/anaconda3/lib/python3.12/site-packages/ultralytics/engine/model.py", line 289, in _load
    self.model, self.ckpt = attempt_load_one_weight(weights)
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/visitor/anaconda3/lib/python3.12/site-packages/ultralytics/nn/tasks.py", line 908, in attempt_load_one_weight
    ckpt, weight = torch_safe_load(weight)  # load ckpt
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/visitor/anaconda3/lib/python3.12/site-packages/ultralytics/nn/tasks.py", line 835, in torch_safe_load
    ckpt = torch.load(file, map_location="cpu")
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/visitor/anaconda3/lib/python3.12/site-packages/ultralytics/utils/patches.py", line 86, in torch_load
    return _torch_load(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/visitor/anaconda3/lib/python3.12/site-packages/torch/serialization.py", line 1326, in load
    with _open_zipfile_reader(opened_file) as opened_zipfile:
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/visitor/anaconda3/lib/python3.12/site-packages/torch/serialization.py", line 671, in __init__
    super().__init__(torch._C.PyTorchFileReader(name_or_buffer))
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
RuntimeError: PytorchStreamReader failed reading zip archive: failed finding central directory

**Reply by glenn-jocher** [reply]
@1ZZZyyyh the error indicates an issue with your dataset path or structure. Ensure the `data` argument points to a valid dataset configuration file (`.yaml`) or a properly structured classification dataset with `train/`, `test/`, and optionally `val/` directories, as outlined in the [classification dataset structure guide](https://docs.ultralytics.com/datasets/classify/). Let me know if you need further assistance!
